crates/autopilot/src/run_loop.rs

```diff
@@ -493,6 +495,34 @@ impl RunLoop {
     }
 }
 
+pub fn log_auction_delta(id: i64, previous: &Option<Auction>, current: &Auction) {
```

Comment:
nit: since we have `infra` now, should we put logging functions there similar to how we put everything in `observe` module of the `driver::infra`?

Reply:
@MartinquaXD really disliked this (and for logging I tend to agree) since we are losing ability to filter logs and verbosity levels by module if everything is in the same module.
I did like the tidiness of it, but I guess we would need a separate observe submodule for each component. Wdyt?

Reply:
Would those concerns be avoided if the keep the actual log call `tracing::debug...` in the `domain` file, while all the data preparation for it and the metric calls are moved to observe submodules?

```
fn log_auction_delta(id: i64, previous: &Option<Auction>, current: &Auction) {
   let (added, removed) = observe::runloop::auction::diff(id, previous, current); // here we prepare the logging data and have metric calls
   tracing::debug!(id, added = ?added, "New orders in auction");
   tracing::debug!(id, removed = ?removed, "Orders no longer in auction");
}
```

Reply:
Yes, but the this wouldn't really be an "observe" method (if you look at the observe module in the driver for instance it literally just contain plain log statements). Maybe making a module specific observe submodule for the runloop is the way to go?